src/main/java/com/google/sps/data/UserDatastoreRepository.java

```diff
@@ -0,0 +1,130 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.User;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+/**
+ * This class implements the match repository using the datastore 
+ */
+public class UserDatastoreRepository implements UserRepository {
+    
+  private Collection<User> allUsers = new ArrayList<>();
+  
+  private static final Gson gson = new Gson();
```

Comment:
It looks like you didn't need Gson after all (or did I miss a call-site)? If not, there's this and the includes here and in the test.

Reply:
Oh yea, not using it. Let me delete it.

Done.